### ISSUE TITLE
check for correct permissions of .ovpn files before parsing

### ISSUE THREAD

**Issue by erikbor** [riboseinc/rvc#65]
.ovpn files can now be readable for 'others':
```
-rw-r--r--  1 root  wheel  4301 Sep  1 14:11 ct.ovpn
```

@jjr840430 could you please implement a check for the correct permissions of .ovpn files. Only accept files that that are readable and writeable by the owner, not readable or writeable by the ```group``` and ```others```.

**Comment by jjr840430** [comment]
@erikbor do I need to check whether .ovpn files are readable/writable by only root user?

**Comment by erikbor** [comment]
@jjr840430 for the ones in /opt/rvc/etc/vpn.d yes please, but not for user owned ones (e.g. in ~/.rvc/)

**Comment by jjr840430** [comment]
@erikbor I think it's very complicated to get owned user from the directory location which has VPN configuration.

**Comment by erikbor** [comment]
I agree that a compromise is not very likely in the current setup but reducing attack surface never hurts. And forcing users to correctly set permissions is also not a bad thing.

**Comment by jjr840430** [comment]
rvd daemon is running as root privilege, so it's not possible to get owned user from '~/.rvc/' at run-time.

**Comment by erikbor** [comment]
Issue [73](https://github.com/riboseinc/rvc/issues/73) is a vulnerability that allows an attacker to execute any executable as gid 0. I understand that there are some conditions that need to be met to successfully exploit it but all it takes is one vulnerability in rvd that compromises the entire system. That's why we need thorough checking of permissions, not following symlinks, input sanitisation etc.

**Comment by jjr840430** [comment]
Yes, You are right.
My wondering is just how rvd may get the user ID from the specified directory such as ~/.rvc/.
You were mentioned 'for the ones in /opt/rvc/etc/vpn.d yes please, but not for user owned ones (e.g. in ~/.rvc/)'.

How about check for only root user?

**Comment by erikbor** [comment]
```
$ cat rvd.json 
{
  "openvpn_bin": "/opt/openvpn/sbin/openvpn",
  "openvpn_root_check": true,
  "ovpn_up_down_scripts": false,
  "user_id": 501,
  "restrict_socket": true,
  "socket_path": "/tmp/.rvd_cmd",
  "log": "/var/log/rvd.log",
  "vpn_config_paths": [
    "/opt/rvc/etc/vpn.d",
    "/usr/local/etc/vpn.d"
  ]
}
```
If user A placed ```ct.json``` and ```ct.ovpn``` in ```/usr/local/etc/vpn.d``` and forgot to set the correct permissions (e.g. ```rwxrwxrwx``` aka ```chmod 777```) then user B could own user A by changing the ```pre-connect-exec``` in ```ct.json```.

And user B could also obtain the private key that is stored in user A's ```ct.ovpn``` if the permissions are set to read for others.

**Comment by jjr840430** [comment]
fixed on #74
